packages/integration-karma/test/shadow-dom/EventTarget-methods/EventTarget.addEventListener.spec.js

```diff
@@ -1,11 +1,118 @@
+            document.body,
+            document.documentElement,
+            document,
+            window,
+        ];
+
+        const log = [];
+        targets.forEach((node) => {
+            node.addEventListener('click', () => log.push(node));
+        });
+
+        nodes.button.click();
+
+        expect(log).toEqual(targets);
+    });
+
+    it('should accept a listener config as second parameter for all nodes except shadow root and host', () => {
+        const targets = [
+            nodes.button,
+            nodes['container_div'],
+            /*
+            TODO [#2134]: These are valid cases but currently fail.
+            nodes['x-container'].shadowRoot,
+            nodes['x-container'],
+            */
+            document.body,
+            document.documentElement,
+            document,
+            window,
+        ];
+
+        const log = [];
+        targets.forEach((node) => {
+            node.addEventListener('click', { handleEvent: () => log.push(node) });
+        });
+
+        nodes.button.click();
+
+        expect(log).toEqual(targets);
+    });
+
+    if (!process.env.COMPAT) {
+        it('should accept a function listener as second parameter for a non-node EventTarget', () => {
+            const target = new EventTarget();
+            const listener = jasmine.createSpy();
+            target.addEventListener('dummy', listener);
+            target.dispatchEvent(new CustomEvent('dummy'));
+            expect(listener).toHaveBeenCalled();
+        });
+
+        it('should accept a listener config as second parameter for a non-node EventTarget', () => {
+            const target = new EventTarget();
+            const listener = jasmine.createSpy();
+            target.addEventListener('dummy', { handleEvent: listener });
+            target.dispatchEvent(new CustomEvent('dummy'));
+            expect(listener).toHaveBeenCalled();
+        });
+    }
+
+    it('should call event listener with the same order', () => {
+        const logs = [];
+        for (let i = 1; i <= 10; i++) {
+            nodes.button.addEventListener('foo', () => logs.push(i));
+        }
+        nodes.button.dispatchEvent(new CustomEvent('foo'));
+        expect(logs).toEqual([1, 2, 3, 4, 5, 6, 7, 8, 9, 10]);
     });
-    elm.dispatchEvent(new CustomEvent('test'));
 
-    expect(handleEvent).toHaveBeenCalled();
+    it('should accept null as second parameter', () => {
+        expect(() => nodes.button.addEventListener('dummy', null)).not.toThrowError();
+    });
+
+    it('should accept undefined as second parameter', () => {
+        expect(() => nodes.button.addEventListener('dummy', undefined)).not.toThrowError();
+    });
+
+    if (!process.env.COMPAT) {
+        // Safari 10 does not throw these errors even though they are part of the spec
+        it('should throw error when second parameter is not passed', () => {
+            expect(() => nodes.button.addEventListener('dummy')).toThrowError();
```

Comment:
Whenever you if a native API throws, I would always recommend asserting the error type:

```suggestion
            expect(() => nodes.button.addEventListener('dummy')).toThrowError(TypeError);
```

Here is why it's important. Most of the JavaScript runtime errors are reported as `Error`. What you want to make sure is that your test is catching the error produced by the browser when validation the arguments and not any runtime error. By being specific about the error type, you reduce the chances of false positives.

Please update the rest of the `toThrowError` with a specific error constructor when possible.